File: coding/java/ReverseString.java
package coding.java;

public class ReverseString {

    public static void main(String[] args) {
        String str = "Hello World !!!";
        System.out.println(str);
        System.out.println("=====Reverse String 111 ======");
        System.out.println(reverseWithStringBuilder(str));
        StringBuffer str1 = new StringBuffer(str);
        System.out.println("===Reverse String 222==" +str1.reverse().toString());
        System.out.println("========= Reverse String 3333 =========");    
        System.out.println(reverseStringManually(str));
    }
    
    private static String reverseWithStringBuilder(String str){
        return new StringBuilder(str).reverse().toString();
    }

    private static String reverseStringManually(String str){
        StringBuilder sb = new StringBuilder();

        for(int i = str.length() - 1; i >= 0; i--){
            sb.append(str.charAt(i));
        }
        return sb.toString();

    }

}

File: coding/python/ReverseString.py
class ReverseString:

    # constructor
    def __init__(self, text):
        self.text = text

    #-----------------------
    # Reverse String Method
    #-------------------------

    def reverse(self):

        # convert string to array/list
        char_array = list(self.text)

        start = 0
        end = len(char_array) - 1

        #Swap Characters

        while start < end:

            char_array[start], char_array[end] = char_array[end], char_array[start]

            start += 1
            end -= 1

        # convert array back to string
        reversed_string = "".join(char_array)

        return reversed_string

# -----------------------
# Main Program
# -------------------------

obj = ReverseString("Hello World !!!!")
result = obj.reverse()
print("Reverse String === ", result)

